pkg/configuration/yaml.go

```diff
@@ -151,10 +153,29 @@ func (w *WorkflowYaml) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := unmarshal(&raw); err != nil {
 		return err
 	}
+	if err := validateWorkflowConfigurationYaml(raw.Configuration); err != nil {
+		return err
+	}
 	*w = WorkflowYaml(raw)
 	return nil
 }
 
+func validateWorkflowConfigurationYaml(config *WorkflowConfigurationYaml) error {
+	if config == nil {
```

Comment:
nil configuration is fine since it means that the user did not specify a `workflow_configuration` in the file. Since this key is optional it is a valid configuration. What we want to do is check that if the user has specified a configuration that all of the keys do exist. So something like

```suggestion
	if config != nil {
    	    if len(config.OnPullRequestPushed == nil {
	    	    return errors.New("workflow_configuration.on_pull_request_pushed is required")
         	}
	       if config.OnPullRequestClosed == nil {
		  return errors.New("workflow_configuration.on_pull_request_closed is required")
        	}
	        if config.OnCommitToDefault == nil {
		    return errors.New("workflow_configuration.on_commit_to_default is required")
        	}
        }
```